### PR TITLE
Interface definition

### PR DESCRIPTION
**Pull request by AdithyahNair**
# Pull Request

## Description

This PR defines the Python interface for our AI Conversation Client based on OpenAI's ChatGPT. It includes class definitions, method signatures, and comprehensive tests for the interface without actual implementation.

Key components:
- `MessageRole` enum for defining message sender roles (user, assistant, system, function)
- `Message` class for representing individual messages with content, role, and metadata
- `Conversation` class for managing collections of messages
- `AIClient` class for interacting with OpenAI's API
- Pytest test suite mocking the implementation to verify interface contracts

This provides a clean foundation for future implementation while clearly defining the API boundaries.

## Type of change

- [ ] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

## How Has This Been Tested?

The interface has been tested using pytest with the following test categories:
- Unit tests for the Message class (creation, properties, default values)
- Unit tests for the Conversation class (messages management, retrieval)
- Unit tests for AIClient (conversation management, message sending/receiving)
- Error case tests for invalid operations

All tests pass successfully by mocking implementation details while verifying interface contracts.

## Checklist

- [x] My code follows the style guidelines of this project (PEP 8 for Python)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (README.md)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my interface design is effective
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by MatthewMa929** [comment]
From Team 1

We found your project to be very well organized and thoughtfully implemented. The README was comprehensive and easy to follow, and the code is clearly documented. The structure of the Message, Conversation, and AIClient classes was well designed, and the functionality provided by each was a strong addition to the overall system.

For improvements, we noticed that timestamps could be used more extensively particularly to record when a conversation is created. Additionally, while your test for get_latest_messages() works well, it would be helpful to include an edge case where the number of messages is fewer than five, to ensure that logic holds under all conditions.

Overall, it was difficult for us to find any major issues. The work was solid, clean, and well thought out. Nice job!

File: ai_conversation_client/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+pytest==7.4.0
```
Comment (mayank-ramnani) [inline ai_conversation_client/requirements.txt]: Use pyproject.toml for specifying dependencies and project configuration.

File: ai_conversation_client/__init__.py [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+            count: The number of messages to retrieve (default: 5)
+            
+        Returns:
+            A list of the most recent messages
+        """
+        return self._messages[-count:] if self._messages else []
+
+
+class AIClient:
+    """Client for interacting with OpenAI's ChatGPT API."""
```
Comment (mayank-ramnani) [inline ai_conversation_client/__init__.py]: This PR is not for implementing a specific backend. This is only for defining the generic interface so that any backend can be implemented on it in the future. 
Remove implementation details and dependency on OpenAI.

File: ai_conversation_client/tests/test_client.py [hunk 3]
```diff
@@ -0,0 +1,213 @@
+"""
+Tests for the AI Conversation Client API.
+
+These tests verify the interface of the AI Conversation Client
+without actually implementing the functionality.
+"""
+
```
Comment (mayank-ramnani) [inline ai_conversation_client/tests/test_client.py:2]: Tests are behaviorally correct, but once you remove the backend dependency and implementation details from the interface definition, they will need to be restructured. 
A dummy implementation can be provided within the test file which will then be used to test the structure of the interface.

**Review by mayank-ramnani — COMMENTED** [review]
Use dependency injection and abstract classes for interface — there should be no backend dependencies. 
Remove extra files from PR (cache files).
Use pyproject.toml.
Make sure there are no ruff or mypy fails before submission.
